File: Sources/XMTPiOS/Libxmtp/DisappearingMessageSettings.swift
//
//  DisappearingMessageSettings.swift
//  XMTPiOS
//
//  Created by Naomi Plasterer on 2/9/25.
//

import LibXMTP

public struct DisappearingMessageSettings {
	public let disappearStartingAtNs: Int64
	public let retentionDurationInNs: Int64
	
	public init(disappearStartingAtNs: Int64, retentionDurationInNs: Int64) {
		self.disappearStartingAtNs = disappearStartingAtNs
		self.retentionDurationInNs = retentionDurationInNs
	}

	static func createFromFfi(_ ffiSettings: FfiMessageDisappearingSettings) -> DisappearingMessageSettings {
		return DisappearingMessageSettings(
			disappearStartingAtNs: ffiSettings.fromNs,
			retentionDurationInNs: ffiSettings.inNs
		)
	}
}

File: sdks/android/shared/src/main/java/org/xmtp/kotlin/libxmtp/DisappearingMessageSettings.kt
package org.xmtp.kotlin.libxmtp

import uniffi.xmtpv3.FfiMessageDisappearingSettings

class DisappearingMessageSettings(
    val disappearStartingAtNs: Long,
    val retentionDurationInNs: Long,
) {
    companion object {
        fun createFromFfi(ffiSettings: FfiMessageDisappearingSettings): DisappearingMessageSettings =
            DisappearingMessageSettings(ffiSettings.fromNs, ffiSettings.inNs)
    }
}
